src/NuGet.Clients/PackageManagement.UI/Models/PackageItemListViewModel.cs

```diff
@@ -265,6 +266,159 @@ internal Lazy<Task<BackgroundLoaderResult>> BackgroundLoader
 
         public Uri IconUrl { get; set; }
 
+        // same URIs can reuse the bitmapImage that we've already used.
+        static Dictionary<Uri, BitmapImage> bitmapImageCache = new Dictionary<Uri, BitmapImage>();
```

Comment:
`private static readonly Dictionary<Uri, BitmapImage> _bitmapImageCache = new Dictionary<Uri, BitmapImage>();` So we never have to worry about null refs?